### ISSUE TITLE
Incorrect image layer validation with secondary command buffers

### ISSUE THREAD

**Issue by karl-lunarg** [KhronosGroup/Vulkan-ValidationLayers#92]
**Issue by [fduranleau-gl](https://github.com/fduranleau-gl) (MIGRATED)**
_Tuesday May 08, 2018 at 13:32 GMT_
_Originally opened as https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/issues/2653_

----

We had this situation where we used an image in a render pass both as a depth attachment and as shader resource. The render pass set the image's layout to VK_IMAGE_LAYOUT_DEPTH_STENCIL_ATTACHMENT_OPTIMAL, but the same image was used in a descriptor with layout VK_IMAGE_LAYOUT_DEPTH_STENCIL_READ_ONLY_OPTIMAL. So obviously, we had image layout conflicts. However, commands for the render pass were recorded into secondary command buffers, and we had no errors being reported by core validation layer. As soon as we did not use secondary command buffers, we got the errors being reported.

There is no need record those secondary command buffes in parallel in other threads to get this error. Simply doing this sequence of commands will show the problem:

1. begin the render pass, with flag to use secondary command buffers
2. begin a secondary command buffer, with inheritance using render pass from 1.
3. issue a draw command using one of the render pass' attachments in a descriptor, but with a different layout
4. end command buffer
5. execute command buffer
6. end render pass

Doing the above, validation reports no errors. But doing this:

1. begin the render pass
2. issue a draw command using one of the render pass' attachments in a descriptor, but with a different layout
3. end render pass

triggers the expected error.

For the record, if it changes something to analyse this problem, the whole render pass in our use case is in its own primary command buffer. The rest is in other primary command buffers.

**Comment by karl-lunarg** [comment]
<a href="https://github.com/tobine"><img src="https://avatars0.githubusercontent.com/u/7526739?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [tobine](https://github.com/tobine) (MIGRATED)**
_Tuesday May 08, 2018 at 13:40 GMT_

----

@jzulauf-lunarg this replaces #2640

**Comment by karl-lunarg** [comment]
Duplicate of #93.  Partial migration of issue caused by restarting GitHub issue migration process interrupted by GitHub abuse detection mechanism.
